oe_theme.theme

```diff
@@ -1242,11 +1242,27 @@ function oe_theme_preprocess_block__system_branding_block(&$variables) {
   $route_options = ['absolute' => TRUE];
   $variables['site_logo_href'] = Url::fromRoute('<front>', [], $route_options)->toString();
 
-  // Ensure the site logo falls back to english version if the language is
-  // not EU language.
+  // Provide correct icon for non-eu languages.
   $language_code = \Drupal::languageManager()->getCurrentLanguage()->getId();
   if (!EuropeanUnionLanguages::hasLanguage($language_code)) {
-    $variables['current_language_id'] = 'en';
+    $language_icon_mapping = [
+      'nb' => 'no',
+      'tr' => 'tu',
+    ];
+    if (isset($language_icon_mapping[$language_code])) {
+      // Set correct icon for Norwegian and Turkish languages.
+      $variables['current_language_id'] = $language_icon_mapping[$language_code];
+    }
```

Comment:
Instead of mapping the language codes here, can we move the logic in the twig filter that currently just returns the exact same language code if it's not an EU language?\Drupal\oe_theme_helper\TwigExtension\TwigExtension::toInternalLanguageId

Reply:
toInternalLanguageId supports European languages only. We can reorganise it as a part of EWPP-1502 but currently it is out of scope. We will be able to use toInternalLanguageId for Norwegian language, but Turkish language uses "tr" language code but ECL team decided to name logo with "tu" suffix that we don't use anywhere. The best option is to ask ECL team to rename Turkish logo with "tr" in logo name but I suppose it will be long. So imho, for now, it is proper solution.